test/index.cpp

```diff
@@ -721,6 +721,110 @@ TEST(lookup, Issue2009)
     ASSERT_ARRAYS_EQ(a, b);
 }
 
+TEST(lookup, SNIPPET_lookup1d)
+{
+    //! [ex_index_lookup1d]
+
+    // input array
+    float in_[5] = {10, 20, 30, 40, 50};
+    af::array in(5, in_);
+
+    // indices to lookup
+    int idx_[3] = {1, 3, 2};
+    af::array idx(3, idx_);
+
+    af::array indexed = af::lookup(in, idx);
+    af_print(indexed);
+    // indexed == { 20, 40, 30 };
+
+
+    // indexes can be out of bounds
+    // indexing past end of array
+    int idx_outofbounds_p_[8] = {4,  5,  6,  7,  8,  9, 10, 11};
+    // and indexing before beginning of array
+    int idx_outofbounds_n_[8] = {0, -1, -2, -3, -4, -5, -6, -7};
+
+    af::array idx_outofbounds_p(8, idx_outofbounds_p_);
+    af::array idx_outofbounds_n(8, idx_outofbounds_n_);
+
+    af::array indexed_out_of_bounds_pos = af::lookup(in, idx_outofbounds_p);
+    af::array indexed_out_of_bounds_neg = af::lookup(in, idx_outofbounds_n);
+    // indexed_out_of_bounds_pos = { 50, 50, 40, 30, 20, 10, 50, 40 }
+    af_print(indexed_out_of_bounds_pos);
+    // indexed_out_of_bounds_neg = { 10, 10, 20, 30, 40, 50, 10, 20 }
+    af_print(indexed_out_of_bounds_neg);
```

Comment:
Prints are not necessary because these snippets aren't going to be executed and you are displaying the value in the comment.